examples/model_selection/plot_grid_search_text_feature_extraction.py

```diff
@@ -0,0 +1,235 @@
+# :class:`~sklearn.model_selection.RandomizedSearchCV`. Using a
+# :class:`~sklearn.model_selection.GridSearchCV` instead would explore all the
+# possible combinations on the grid, which can be costly to compute, whereas the
+# parameter `n_iter` of the :class:`~sklearn.model_selection.RandomizedSearchCV`
+# controls the number of different random combination that are evaluated. Notice
+# that setting `n_iter` larger than the number of possible combinations in a
+# grid would lead to repeating already-explored combinations.
+
+import numpy as np
+
+parameters = {
+    "vect__max_df": (0.2, 0.4, 0.6, 0.8, 1.0),
+    "vect__min_df": (1, 3, 5, 10),
+    "vect__ngram_range": ((1, 1), (1, 2)),  # unigrams or bigrams
+    "vect__norm": ("l1", "l2"),
+    "clf__alpha": np.logspace(-6, 6, 13),
+}
+
+# %%
+# We search for the best parameters for both the feature extraction and the
+# classifier. In this case `n_iter=40` is not an exhaustive search of the
+# hyperparameter grid. In practice it would be interesting to increase the
+# parameter `n_iter` to get a more informative analysis. The consequent increase
+# in computing time can be handled by increasing the number of CPUs via the
+# `n_jobs` parameter.
+
+from pprint import pprint
+from sklearn.model_selection import RandomizedSearchCV
+
+random_search = RandomizedSearchCV(
+    estimator=pipeline,
+    param_distributions=parameters,
+    n_iter=40,
+    random_state=0,
+    n_jobs=2,
+    verbose=1,
+)
+
+print("Performing grid search...")
+print("parameters:")
+pprint(parameters)
+
+# %%
+from time import time
+
+t0 = time()
+random_search.fit(data_train.data, data_train.target)
+print(f"done in {time() - t0:.3f}s")
+
+# %%
+print("Best parameters set:")
+best_parameters = random_search.best_estimator_.get_params()
+for param_name in sorted(parameters.keys()):
+    print(f"{param_name}: {best_parameters[param_name]}")
+
+# %%
+test_accuracy = random_search.score(data_test.data, data_test.target)
+print(
+    "Accuracy of the best parameters using the inner CV of "
+    f"the random search: {random_search.best_score_:.3f}"
+)
+print(f"Accuracy on test set: {test_accuracy:.3f}")
+
+# %%
+# The prefixes `vect` and `clf` are required to avoid possible ambiguities in
+# the pipeline, but are not necessary for visualizing the results. Because of
+# this, we define a function that will rename the tuned hyperparameters and
+# improve the readability.
+
+import pandas as pd
+
+
+def shorten_param(param_name):
+    if "__" in param_name:
+        return param_name.rsplit("__", 1)[1]
+    return param_name
+
+
+cv_results = pd.DataFrame(random_search.cv_results_)
+cv_results = cv_results.rename(shorten_param, axis=1)
+
+# %%
+# We can use a `plotly.express.scatter
+# <https://plotly.com/python-api-reference/generated/plotly.express.scatter.html>`_
+# to visualize the trade-off between scoring time and mean test score. Passing
+# the cursor over a given point displays the corresponding parameters.
+
```

Comment:
```suggestion
# to visualize the trade-off between scoring time and mean test score
# (i.e. "CV score"). Passing the cursor over a given point displays the
# corresponding parameters.
```

Reply:
We also need to add the information regarding the error bar. It is not so common to use the scatter plot with error bars.
It should be better to place it before the figure to not be surprise.